pkg/resources/reconciling/ensure.go

```diff
@@ -60,6 +62,49 @@ func createWithName(rawcreate ObjectCreator, name string) ObjectCreator {
 	}
 }
 
+// EnsureObject is specification of the object to ensure.
+type EnsureObject struct {
+	Name             types.NamespacedName
+	Creator          ObjectCreator
+	EmptyObj         runtime.Object
+	RequiresRecreate bool
+}
+
+// EnsureNamedObjectsConcurrent calls `EnsureNamedObject` for each specifiecation concurrently and returns aggregated errors if any occurred.
+func EnsureNamedObjectsConcurrent(ctx context.Context, client ctrlruntimeclient.Client, objs []EnsureObject) error {
+	var wg sync.WaitGroup
+	errChan := make(chan error)
+	for _, obj := range objs {
+		wg.Add(1)
+		go func(obj EnsureObject) {
+			defer wg.Done()
+			err := EnsureNamedObject(ctx, obj.Name, obj.Creator, client, obj.EmptyObj, obj.RequiresRecreate)
+			if err != nil {
+				errChan <- fmt.Errorf("ensure %+v: %v", obj.Name, err)
```

Comment:
```suggestion
				errChan <- fmt.Errorf("ensure %+v: %w", obj.Name, err)
```

wrap the error with "new" (as of go1.13) error handling

Reply:
AFAIK %w is useful when client wants to [unwrap the error](https://golangbyexample.com/wrapping-and-unwrapping-error-golang/) and check the type/kind of the error for some special case. Because `EnsureNamedObject` uses `fmt.Errorf` I would change there to %w too. But that means the client would need to unwrap at least twice to get to some error he can check against. 
Sometimes to ensure clients do not use unwrap %v is used intentionally. Do you think it could be what we want?